channel/player.go

```diff
@@ -773,17 +840,61 @@ func (d *player) removeItem(item item) {
 	d.send(packetInventoryRemoveItem(item))
 }
 
+func (d *player) dropMesos(amount int32) error {
+	if d.mesos < amount {
+		return errors.New("not enough mesos")
+	}
+
+	d.takeMesos(amount)
+
+	return nil
+}
+
+func (d *player) pickupItem(pos pos, dropID int32) {
```

Comment:
The logic in this function is better placed in the handler and doesn't need to be a method.